beacon-chain/blockchain/process_block.go

```diff
@@ -76,6 +76,8 @@ func (s *Service) postBlockProcess(cfg *postBlockProcessConfig) error {
 
 	err := s.cfg.ForkChoiceStore.InsertNode(ctx, cfg.postState, cfg.roblock)
 	if err != nil {
+		// Do not use parent context in the event it deadlined
+		ctx = context.Background()
```

Comment:
This will lose the span information which breaks tracing on this routine.

Consider doing this:
```go
ctx := trace.NewContext(context.Background(), span)
```

This will create a new context without deadlines and load the metadata from the span on this routine onto that new context.